frame/nomination-pools/src/lib.rs

```diff
@@ -535,6 +535,39 @@ impl<T: Config> PoolMember<T> {
 		}
 	}
 
+	/// Total balance of the member, both active and unbonding.
+	/// Doesn't mutate state.
+	#[cfg(any(feature = "try-runtime", test))]
+	fn total_balance(&mut self) -> BalanceOf<T> {
+		if let Some(pool) = BondedPool::<T>::get(self.pool_id).defensive() {
+			let active_balance = pool.points_to_balance(self.active_points());
+
+			// get all unbonding balance, both already unlocked and yet to be unlocked.
+			match SubPoolsStorage::<T>::get(self.pool_id) {
+				Some(mut sub_pools) => {
+					let unbonding_balance = &self.unbonding_eras.iter().fold(
+						BalanceOf::<T>::zero(),
+						|accumulator, (era, unlocked_points)| {
+							if let Some(era_pool) = sub_pools.with_era.get_mut(era) {
+								accumulator.saturating_add(
+									era_pool.point_to_balance(unlocked_points.clone()),
+								)
+							} else {
+								accumulator.saturating_add(
+									sub_pools.no_era.point_to_balance(unlocked_points.clone()),
+								)
+							}
+						},
+					);
+					active_balance.saturating_add(unbonding_balance.clone())
+				},
+				None => active_balance,
+			}
+		} else {
+			Zero::zero()
+		}
+	}
```

Comment:
This has way too many levels of indentation making it difficult to understand.

Could you please add some comments and also see if you can reduce the levels of indentation?

Reply:
Thanks for pointing it out. It's rewritten and also a comment added to the selection of the `SubPool`.